### PR TITLE
Block Bindings: Fix showing bindings field values in theme templates

### PR DESCRIPTION
**Pull request by SantosGuillamot**
## What?
While working on adding [this test](https://github.com/WordPress/gutenberg/pull/65526/files#diff-169d4acdecb966fa33b188dcf2bfa92aca9f3ac66c188a4b8facbd45b5e806dcR62), I realized that connected blocks in CPT templates like `single-movie` weren't showing the default values. When you create a single-movie template from the editor, it works as expected. However, when you directly create a `single-movie.html` in your theme, it doesn't.

This pull request aims to solve that.

## Why?
It should work for theme files as well.

## How?
After triaging it, it seems that `single-movie.html` is treated as a custom template, even if it is specific to movies. For that reason, I've change the conditional to not rely on `post.is_custom` and directly check the slug. That conditional was there to cover use cases like `page-custom-template`. With the new code, it directly checks that the theme slug **matches** "page", while previously it was checking the theme slug **starts with** "page". 

## Testing Instructions
This one is hard to test manually because you need a CPT theme template like `single-movie.html` and cannot be created through the editor UI. It will be covered by [this automated test](https://github.com/WordPress/gutenberg/pull/65526/files#diff-169d4acdecb966fa33b188dcf2bfa92aca9f3ac66c188a4b8facbd45b5e806dcR62), among others, in the future.

Basically, you would need to:
1. Register a custom field for a "movie" CPT:
register_meta(
		'post',
		'movie_title',
		array(
			'label'          => 'Movie Title',
			'object_subtype' => 'movie',
			'show_in_rest' => true,
			'single'       => true,
			'type'         => 'string',
			'default'	   => 'Default movie title',
		)
	);
2. Create a `single-movie.html` file in your theme.
3. Go to your single movie template and add a paragraph connected to that field.
4. Check that it shows the default value instead of the key.

### CODE REVIEW

File: packages/editor/src/components/provider/index.js [hunk 1]
```diff
@@ -188,26 +188,20 @@ export const ExperimentalEditorProvider = withRegistryProvider(
...
-							const match = post.slug.match(
-								`^single-(${ postTypesSlugs.join(
-									'|'
-								) })(?:-.+)?$`
-							);
-							if ( match ) {
-								postContext.postType = match[ 1 ];
-							}
-							break;
+				const themeSlug = post.slug;
```
Comment (cbravobernal) [inline packages/editor/src/components/provider/index.js]: I know this case is for theme templates. But still I don't really like the naming.
Maybe we could just use `post.slug` on the comparisons.

**Review by cbravobernal — APPROVED** [review]
Just added a nitpick.

File: packages/editor/src/components/provider/index.js (hunk 1 above)
Comment (SantosGuillamot) [reply packages/editor/src/components/provider/index.js]: Changed in this commit 🙂 https://github.com/WordPress/gutenberg/pull/65639/commits/3344a5b3bc6cfc24bf0fa631ba913954765c6240

File: packages/editor/src/components/provider/index.js [hunk 2]
```diff
@@ -188,26 +188,19 @@ export const ExperimentalEditorProvider = withRegistryProvider(
...
-							break;
+				if ( post.slug === 'page' ) {
+					postContext.postType = 'page';
+				} else if ( post.slug === 'single' ) {
+					postContext.postType = 'post';
+				} else if ( post.slug.split( '-' )[ 0 ] === 'single' ) {
+					// If the slug is single-{postType}, infer the post type from the slug.
+					const postTypesSlugs =
+						postTypes?.map( ( entity ) => entity.slug ) || [];
+					const match = post.slug.match(
```
Comment (youknowriad) [inline packages/editor/src/components/provider/index.js:199]: I noticed that post.slug is not part of the dependencies of the useMemo call which means potentially a hidden bug.

Comment (SantosGuillamot) [reply packages/editor/src/components/provider/index.js:199]: Thanks for pointing it out. I started [this pull request](https://github.com/WordPress/gutenberg/pull/66125) to address that.
